### PR TITLE
ENH: add CLI flag to force calling Setup::InitFlow ahead of dump reading at restart

### PR DESCRIPTION
**Pull request by neutrinoceros**
As discussed earlier this week with @glesur, add a way to force initial conditions *ahead* of dump reading on restart. This allows users to add new datastructures on restart (like particles, multi-fluids ...).

TODO:
- [x] validate that this is a sound approach for particles (this will be done externally, on my private fork)
- [x] agree on the exact name of the flag
- [ ] discuss the limits of this implementation and wether this is actually needed or not in all the cases we mean to support
- [x] add documentation for it

### CODE REVIEW

**Comment by neutrinoceros** [comment]
weird, looks like I somehow skipped CI when I opened this PR. I expect to push more changes later so hopefully it'll trigger then.

File: src/main.cpp [hunk 1]
```diff
@@ -111,6 +111,11 @@ int main( int argc, char* argv[] ) {
     ///////////////////////////////
     // Are we restarting?
     if(input.restartRequested) {
+      if(input.forceInitRequested) {
+        idfx::pushRegion("Setup::Initflow");
+        mysetup.InitFlow(data);
+        idfx::popRegion();
+      }
```
Comment (glesur) [inline src/main.cpp:118]: Should also call `DeriveVectorPotential()` otherwise this won't be compatible with VECTOR_POTENTIAL

Comment (neutrinoceros) [reply src/main.cpp:119]: done

File: src/input.cpp [hunk 2]
```diff
@@ -167,6 +167,8 @@ void Input::ParseCommandLine(int argc, char **argv) {
       }
       this->maxCycles = std::stoi(std::string(argv[++i]));
       inputParameters["CommandLine"]["maxCycles"].push_back(std::to_string(maxCycles));
+    } else if(std::string(argv[i]) == "-force_init") {
+      this->forceInitRequested = true;
```
[the remark this answers is not in the document]
Comment (neutrinoceros) [reply src/input.cpp:171]: done

**Pull request merged**